src/hooks/useRowMenu.ts

```diff
@@ -0,0 +1,105 @@
+ *
+ * @category Hooks
+ * @returns A tuple of three values:
+ * - operationsRef - assign this reference to `<RowOperationsButton />`
+ * - parentRef - assign this reference to common parent of `<RowOperationsButton />` and `<Table />`
+ * - handleRowHover - `onMouseEnter`/`onMouseLeaver` pair for antd `<Table onRow />` property
+ */
+export function useRowMenu() {
+    const operationsRef = React.useRef<RowOperationsButtonInstance>()
+    const parentRef = React.useRef()
+    const handleRowHover = (record: DataItem) => ({
+        onMouseEnter: (e: React.MouseEvent<HTMLTableRowElement>) => {
+            operationsRef.current?.setRow?.(record, e)
+        },
+        onMouseLeave: (e: React.MouseEvent<HTMLTableRowElement>) => {
+            operationsRef.current?.setRow?.(null, e)
+        }
+    })
+    return [operationsRef, parentRef, handleRowHover] as const
+}
+
+/**
+ * Extends `ref` instance with `setRow` method which can be used to hover row
+ *
+ * When hovered, `parent` element is placed in the middle of hovered row.
+ *
+ * @param ref - Reference to `<RowOperationsButton />` instance
+ * @param parent - Common parent for `<RowOperationsButton />` and `<Table />`
+ * @param container - Root element of `<RowOperationsButton />`
+ * @param skip - Condition to skip hover
+ * @param onRowHover - Callback for hovered record id
+ * @param onRowLeave - Callback when leaving hovered row
+ */
+export function useRowMenuInstance(
+    ref: React.ForwardedRef<RowOperationsButtonInstance>,
+    parent: MutableRefObject<HTMLElement>,
+    containerRef: MutableRefObject<HTMLElement>,
+    skip: boolean,
+    onRowHover: (id: string) => void,
+    onRowLeave: () => void
+) {
+    return React.useImperativeHandle(
+        ref,
+        () => ({
+            containerRef,
+            setRow: (record, e) => {
+                const initialized = !!(parent.current && containerRef.current)
+
+                if (skip || !initialized) {
+                    return
+                }
+                const row = record ? e.currentTarget : null
+                if (!row && isOutsideMove(e, parent.current)) {
+                    onRowLeave()
+                } else if (row && containerRef.current && parent.current) {
+                    containerRef.current.style.display = 'block'
+                    const { top: rowTop } = row.getBoundingClientRect()
+                    const { top: tableTop } = parent.current.getBoundingClientRect()
+                    /**
+                     * To position in the middle:
+                     *
+                     * rowTop - tableTop + height / 2 - buttonSize / 2
+                     */
+                    containerRef.current.style.top = `${rowTop - tableTop + 17}px`
+                    onRowHover(record.id)
+                }
+            }
+        }),
+        [parent?.current, containerRef?.current, skip, onRowHover, onRowLeave]
```

Comment:
Apparently this is intentional

1) Why destructuring for 1 variable?
2) Extracting ref from dependency array outside of the `useImperativeHandle` [will break](https://github.com/tesler-platform/tesler-ui/actions/runs/785677799) the behavior. Refs are mutable, hence they do not cause rerender when `current` is changed, hence `useRowMenu` will not be called again when its parent changes refs `current`, hence both container and parent will be undefined and will never be initialized.
Current behavior works because it's a callback, and actual `current` value is accessed during user interaction. My understanding was that refs are never should be part of dependencies arrays at all (as `ref` itself is consistent, and `ref.current` not causes rerenders); extracting `current` to separate variable is not a correct way.

I suppose we'll need to do some reading about how eslint plugin works, if we should remove all refs from all dependencies or there is a possibility for stale state and we should change our approach with their usage.